### PR TITLE
Check plot description size in advance

### PR DESCRIPTION
**Pull request by i1i1**
Following suggestion of @nazar-pc of how to check minimal plot size

### CODE REVIEW

**Review by ozgunozerk — APPROVED** [review]

File: src/farmer.rs [hunk 1]
```diff
@@ -86,37 +87,45 @@ impl CacheDescription {
...
     pub space_pledged: ByteSize,
 }
 
+/// Error type for cache description constructor
+#[derive(Debug, Clone, Copy, thiserror::Error)]
+#[error("Cache should be larger than {}", PlotDescription::MIN_SIZE)]
+pub struct PlotConstructionError;
+
 impl PlotDescription {
+    const SECTOR_OVERHEAD: ByteSize = ByteSize::mb(1);
```
[the remark this answers is not in the document]
Comment (i1i1) [reply src/farmer.rs:106]: How can I calculate the prefix size though?

[the remark this answers is not in the document]
Comment (nazar-pc) [reply src/farmer.rs:106]: It is a fixed size reserved space to allow for metadata upgrades in the future:
https://github.com/subspace/subspace/blob/d06ad0bad8a9cac203d0e0a9b27611b9f0f48e2a/crates/subspace-farmer/src/single_disk_plot.rs#L56-L57

[the remark this answers is not in the document]
Comment (i1i1) [reply src/farmer.rs:106]: Okay. I'll add a todo to account for that. Thanks!

File: src/farmer.rs [hunk 2]
```diff
@@ -57,20 +57,21 @@ impl CacheDescription {
...
+    /// Minimal cache size
+    pub const MIN_SIZE: ByteSize = ByteSize::mib(1);
+
     /// Construct Plot description
     pub fn new(
         directory: impl Into<PathBuf>,
         space_dedicated: ByteSize,
     ) -> Result<Self, CacheTooSmall> {
-        if space_dedicated < MIN_CACHE_SIZE {
+        if space_dedicated < Self::MIN_SIZE {
```
[the remark this answers is not in the document]
Comment (i1i1) [reply src/farmer.rs:74]: I double checked, and it seems I have a bug here. I returned `Ok` only on plots which are smaller than `Self::MIN_SIZE`.

Fixed it now.

File: src/farmer.rs [hunk 3]
```diff
@@ -86,37 +87,45 @@ impl CacheDescription {
 }
 
 /// Description of the plot
```
Comment (nazar-pc) [inline src/farmer.rs:89]: BTW, there is a data structure with details about the plot in farmer already and it has more details in it.

**Review by nazar-pc — APPROVED** [review]
Not sure why cache size has minimum amount of space though, looks a bit strange to me

File: src/farmer.rs (hunk 3 above)
Comment (i1i1) [reply src/farmer.rs:89]: Yeah, but this is just a description. The same thing as this:

https://github.com/subspace/subspace/blob/main/crates/subspace-farmer/src/bin/subspace-farmer/main.rs#L113-L119

**Pull request merged**